### PR TITLE
[CS-2983]: Add initial tabBar layout as feature flag

### PR DESCRIPTION
**Pull request by danibonilha**
### Description

This PR adds the tabbar with a feature flag since we need to map internal flows and behaviors in order to replace the current navigation style, it's mostly design, don't expect to navigate correctly in inner flows.

### Screenshots

Screen Shot 2022-01-20 at 16 50 59

iOS:

Simulator Screen Recording - iPhone 11 - 2022-01-20 at 16 51 24

Android:
android-tabbar

### CODE REVIEW

File: cardstack/src/navigation/tabBarNavigator.tsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+}
+
+const TabBarFeatureContext = createContext<TabBarContextType>({
+  isTabBarEnabled: false,
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  setIsTabBarEnabled: () => {},
+});
+
+export const TabBarFeatureProvider: React.FC = ({ children }) => {
+  const [isTabBarEnabled, setIsTabBarEnabled] = useState(false);
```
Comment (danibonilha) [inline cardstack/src/navigation/tabBarNavigator.tsx:110]: Set this to true to enable the tab if you want to check it out

File: cardstack/src/navigation/tabBarNavigator.tsx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+      isTabBarEnabled,
+      setIsTabBarEnabled,
+    }),
+    [isTabBarEnabled]
+  );
+
+  return (
+    <TabBarFeatureContext.Provider value={contextValues}>
+      {isTabBarEnabled ? <StackNavigator /> : children}
+    </TabBarFeatureContext.Provider>
```
Comment (danibonilha) [inline cardstack/src/navigation/tabBarNavigator.tsx:123]: We haven't yet decided how to enable the feature in a release, or internal build, but I created as a context to make it easier to apply when we do decide

**Review by topwebtek7 — COMMENTED** [review]
Looks good to me! @danibonilha what about we upgrade React-navigation to latest version(v6) when change to TabBar... I think it could fix QRCode scanner blackout things sometimes happens(cuz of `useFocusEffect`'s not stable) And if we can, get rid of react-native-cool-modals would be perfect, IMO.

**Review by lukemelia — APPROVED** [review]
I tried it out on iOS and the tab bar and interaction look good 👍

**Review by oidoug — APPROVED** [review]
Looks good on both ios and android.
I feel the wallet should be in the middle.

**Comment by danibonilha** [comment]
> Looks good to me! @danibonilha what about we upgrade React-navigation to latest version(v6) when change to TabBar... I think it could fix QRCode scanner blackout things sometimes happens(cuz of `useFocusEffect`'s not stable) And if we can, get rid of react-native-cool-modals would be perfect, IMO.

I thought about migrating to v6 but as of today there are still lots of diff navigation configs, my thought is to simplify the nav first, then upgrade to the latest version, it's on the radar.

**Pull request merged**